pkg/resource/v3/resource.go

```diff
@@ -25,6 +25,9 @@ const (
 	ExtensionConfigType = APITypePrefix + "envoy.config.core.v3.TypedExtensionConfig"
 	RuntimeType         = APITypePrefix + "envoy.service.runtime.v3.Runtime"
 
+	// Rate Limit service
+	RateLimitConfigType = APITypePrefix + "ratelimit.config.ratelimit.v3.RateLimitConfig"
```

Comment:
Do we not need to add anything in the `server/` folder to support sending configuration over the grpc streams? This is only in the cache which adds support there but is the intention to send RLS configs over an ADS stream?

Forgive me its been a while since I've looked into the ratelimiting subsystem so if Im missing something obvious let me know. But if the former is the use case target we will need to add the functionality inside the gRPC server alongside the cache.

Reply:
I see that you have this outlined in one of your previous issues:
```proto
// [#protodoc-title: RateLimitConfigDiscoveryService]
service RateLimitConfigDiscoveryService {
    rpc StreamDescriptors(stream envoy.service.discovery.v3.DiscoveryRequest)
        returns (stream envoy.service.discovery.v3.DiscoveryResponse) {
    }
    
    rpc FetchConfigs(envoy.service.discovery.v3.DiscoveryRequest) returns (envoy.service.discovery.v3.DiscoveryResponse) {
    }
}
```
This is what im alluding to

Reply:
> But if the former is the use case target we will need to add the functionality inside the gRPC server alongside the cache.

In the begging, we decided to have a separate RateLimitConfigDiscoveryService. And later based on @mattklein123's suggestion: https://github.com/envoyproxy/ratelimit/pull/368#issuecomment-1281098702 we are going to use ADS stream. If we are going with this approach could you please point me to the functionalities you mentioned above?

Reply:
So yes we will want to be use ADS, but in the go-control-plane repo we still have to implement the service in the `server/` code. Once it's implemented it can be multiplexed over the ADS stream handler found here: https://github.com/envoyproxy/go-control-plane/blob/main/pkg/server/v3/server.go#L300

Note that the other xDS services have similar style implementations but can all be served over ADS.

Reply:
As I understand your comment, I think I have to add `RateLimitConfigDiscoveryService` back. Please correct me if I am wrong.

```proto
// [#protodoc-title: RateLimitConfigDiscoveryService]
service RateLimitConfigDiscoveryService {
    rpc StreamDescriptors(stream envoy.service.discovery.v3.DiscoveryRequest)
        returns (stream envoy.service.discovery.v3.DiscoveryResponse) {
    }
    
    rpc FetchConfigs(envoy.service.discovery.v3.DiscoveryRequest) returns (envoy.service.discovery.v3.DiscoveryResponse) {
    }
}
```

Could you please have a look at the commit https://github.com/envoyproxy/go-control-plane/pull/598/commits/bcfa6c3495909221ba33bed473201f06add8e4b1 in this PR? I removed `RateLimitConfigDiscoveryService`, since I was thought to use ADS only. So I think I have to rollback this commit.

Reply:
@renuka-fernando yes you'll want to roll back that commit. We need those so ADS can actually hook into them. ADS simply calls out to the xDS endpoints over a single stream. The control plane still needs to register those gRPC endpoints regardless of the protocol type.

Reply:
Added back with https://github.com/envoyproxy/go-control-plane/pull/598/commits/b4ea9af5085a55a1cf817fecc881dabc8b683daa